### DISCUSSION TITLE
Code simplify

### DISCUSSION THREAD

**Discussion by lotuslakestudio** [DioxusLabs/dioxus#2156] [Q&A]
How to simpify the following code to use **Someelement{"message"}** instead of **Someelement{value:"message"}**?
====================================
```rust
#[component]
pub fn Someelement(value:String)->Element{
    rsx!{{value}}
}

#[component]
pub fn App()->Element{
    rsx!{
        Someelement{value:"message"}
    }
}

fn main() {
    launch(App);
}
```
====================================
Thanks for any response.

**Comment by ealmloff** [comment]
You can use the `children` field (documented [here](https://dioxuslabs.com/learn/0.5/reference/component_props#component-children)) to make Someelement accept any rsx elements as children.
```rust
#[component]
pub fn Someelement(children: Element)->Element{
    rsx!{{value}}
}
```

**Comment by lotuslakestudio** [comment]
Thanks a lot. But how to deal with the following one which is a bit more complex?
```
use dioxus::prelude::*;

#[component]  
pub fn Welcome(name:String, index:i64) -> Element {
    rsx!{"Welcome {name} who has been here for {index-1} times before."}  
}  
  
#[component]  
pub fn App() -> Element {  
    rsx!{  
        Welcome{name:"Lotus",index:8}
    }  
}  
  
fn main() {  
    launch(App);  
}
```
I want to use **Welcome{"Lotus",8}** instead of **Welcome{name:"Lotus",index:8}**

**Comment by ealmloff** [comment]
You cannot do that with dioxus. Components are more like structs than functions. If you have no state or hooks in your "component" you can call it like a normal function, but keep in mind that it will not be memorized and cannot use hooks
